### ISSUE TITLE
Borders appear only for the first bar in the bar plot.

### ISSUE THREAD

**Issue by guziy** [matplotlib/matplotlib#9820]
### Bug report
Only the first bar in barplot has border when edgecolor is a single value.

**Bug summary**

**Code for reproduction**

```python
import matplotlib.pyplot as plt


def main():

    fig = plt.figure()
    ax = plt.gca()

    data = list(range(5, 10))

    ax.bar(data, data, width=0.5, align="edge", linewidth=2,
           edgecolor="k", facecolor="m")

    plt.show()


if __name__ == '__main__':
    main()
```

**Actual outcome**
![screen shot 2017-11-20 at 15 50 34](https://user-images.githubusercontent.com/900941/33040920-97d81072-ce0a-11e7-8bcb-1f5b24f0934b.png)

**Expected outcome**

![screen shot 2017-11-20 at 15 51 51](https://user-images.githubusercontent.com/900941/33040975-c655e05a-ce0a-11e7-9709-8b977428aa5f.png)

I can work around the issue by specifying a list of edgecolors and linewidths for each bar. Specifying only the list of linewidths for each bar and leaving one value for the edgecolor would only show a border of the first bar due to this code in the Patch.draw:
```python
        lw = self._linewidth
        if self._edgecolor[3] == 0:
            lw = 0
```

**Matplotlib version**

  * Operating system: Linux
  * Matplotlib version: 2.1.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: Python 3.6.2
 

I installed matplotlib using conda. But I am not sure from which channel.

**Comment by guziy** [comment]
I could try to fix it if someone confirms that this is not intended behavior))

Cheers

**Comment by jklymak** [comment]
It’s a duplicate report and fixed in master

**Comment by guziy** [comment]
Ah, sorry))

Closing it then

**Comment by jklymak** [comment]
#9351, #9818, #9808 fixed by #9353 ;-)
